app/models/user.rb

```diff
@@ -10,38 +10,28 @@ class User < ApplicationRecord
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
   has_many :friendships
-  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: :friend_id
 
-  # to find all users
-  def friends
-    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
-    friends_array += inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+  def sent_invitation
+    friends_array = friendships.map(&:friend)
     friends_array.compact
   end
 
-  # Users who have yet to confirme friend requests
-  def pending_friends
-    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
-  end
-
-  # Users who have requested to be friends
-  def friend_requests
-    inverse_friendships.map { |friendship| friendship unless friendship.confirmed }.compact
+  def received_invitation
+    friends_array = inverse_friendships.map(&:user)
+    friends_array.compact
   end
 
-  def friend_requestsu
-    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+  def confirmed_friend?(user)
+    sent_invitation.include?(user) && received_invitation.include?(user)
   end
```

Comment:
**Optional**
```ruby
has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships
```
It will be nice to present it in SQL query like :point_up:. Since database was created to store and query data it is always more efficient than ruby.